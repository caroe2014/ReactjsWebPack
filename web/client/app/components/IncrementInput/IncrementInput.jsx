// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { Flex } from 'rebass';
import styled from 'styled-components';
// import IconButton from '../../components/IconButton/IconButton';

const Container = styled(Flex)`
  align-items: center;
  background-color: ${props => props.theme.colors.light};
  border: 0.5px solid ${props => props.theme.colors.primaryText};
  border-radius: 20px;
  justify-content: center;
  padding: 4px;
`;

const Input = styled(Flex)`
  color: ${props => props.theme.colors.primaryText};
  justify-content: center;
  padding: 0 5px;
  min-width: 45px;
`;
// TODO: Fix this after order-service supports quantity requests
// const Button = styled(IconButton)`
//   color: ${props => props.theme.colors.primaryText};
//   font-weight: 400;
//   margin: 0 10px 0 10px
// `;

export default function IncrementInput (props) {
  const { decrement, increment, ...rest } = props;
  return (
    <Container {...rest} id='IncrementInput'>
      {/* TODO: Fix this after order-service supports quantity requests */}
      {/* <Button
        onClick={() => props.decrement()}
        children='&minus;'
      />
      <Input>
        {props.value}
      </Input>
      <Button
        onClick={() => props.increment()}
        children='&#43;'
      /> */}
      <Input style={{ fontWeight: '100', padding: '0px 27px' }}>
        Quantity {props.value}
      </Input>
    </Container>
  );
}
