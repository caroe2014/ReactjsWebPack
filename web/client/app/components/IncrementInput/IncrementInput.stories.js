// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Provider } from 'rebass';
import theme from 'web/client/theme.js';
import IncrementInput from './IncrementInput';

storiesOf('Input', module)
  .add('IncrementInput', () =>
    <Provider theme={theme} style={{ width: '100px' }}>
      <IncrementInput onClick={action('clicked')}/>
    </Provider>
  );
