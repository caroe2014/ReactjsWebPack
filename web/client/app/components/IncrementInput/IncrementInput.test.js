// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import IncrementInput from './IncrementInput';
configure({
  adapter: new Adapter()
});

describe('IncrementInput', () => {

  it('does a snapshot test', () => {
    const wrapper = shallow(<IncrementInput/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<IncrementInput />).exists()).toBe(true);
  });
});
