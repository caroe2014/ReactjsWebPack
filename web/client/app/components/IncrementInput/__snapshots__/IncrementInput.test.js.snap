// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncrementInput does a snapshot test 1`] = `
<Styled(Flex)
  id="IncrementInput"
>
  <Styled(Flex)
    style={
      Object {
        "fontWeight": "100",
        "padding": "0px 27px",
      }
    }
  >
    Quantity 
  </Styled(Flex)>
</Styled(Flex)>
`;
